### ISSUE TITLE
kubeadm init hung at control plane and kubelet is running

### ISSUE THREAD

**Issue by pbedadham** [kubernetes/kubernetes#58582]
/kind bug

**What happened**:kubeadm init  --kubernetes-version=v1.9.2 --apiserver-advertise-address=10.232.219.172 
kubeadm init  --kubernetes-version=v1.9.2 --apiserver-advertise-address=10.232.219.172
[init] Using Kubernetes version: v1.9.2
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks.
	[WARNING FileExisting-crictl]: crictl not found in system path
[preflight] Starting the kubelet service
[certificates] Generated ca certificate and key.
[certificates] Generated apiserver certificate and key.
[certificates] apiserver serving cert is signed for DNS names [centos-4.k8s.com kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.232.219.172]
[certificates] Generated apiserver-kubelet-client certificate and key.
[certificates] Generated sa key and public key.
[certificates] Generated front-proxy-ca certificate and key.
[certificates] Generated front-proxy-client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "scheduler.conf"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
[init] This might take a minute or longer if the control plane images have to be pulled
-----------------
ystemctl status kubelet -l
● kubelet.service - kubelet: The Kubernetes Node Agent
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/kubelet.service.d
           └─10-kubeadm.conf
   Active: active (running) since Sun 2018-01-21 09:42:30 UTC; 3min 25s ago
     Docs: http://kubernetes.io/docs/
 Main PID: 4857 (kubelet)
   Memory: 43.3M
   CGroup: /system.slice/kubelet.service
           └─4857 /usr/bin/kubelet --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true --network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin --cluster-dns=10.96.0.10 --cluster-domain=cluster.local --authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt --cadvisor-port=0 --cgroup-driver=cgroupfs --rotate-certificates=true --cert-dir=/var/lib/kubelet/pki --fail-swap-on=false

Jan 21 09:45:12 centos-4.k8s.com kubelet[4857]: E0121 09:45:12.443733    4857 kubelet.go:2105] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
Jan 21 09:45:14 centos-4.k8s.com kubelet[4857]: E0121 09:45:14.400063    4857 kubelet_node_status.go:106] Unable to register node "centos-4.k8s.com" with API server: Timeout: request did not complete within allowed duration
Jan 21 09:45:14 centos-4.k8s.com kubelet[4857]: E0121 09:45:14.400147    4857 event.go:200] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"centos-4.k8s.com.150bc9f44190cb0a", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"Node", Namespace:"", Name:"centos-4.k8s.com", UID:"centos-4.k8s.com", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"NodeHasSufficientDisk", Message:"Node centos-4.k8s.com status is now: NodeHasSufficientDisk", Source:v1.EventSource{Component:"kubelet", Host:"centos-4.k8s.com"}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xbe9136019acb0f0a, ext:172105729, loc:(*time.Location)(0x58992c0)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xbe9136019acb0f0a, ext:172105729, loc:(*time.Location)(0x58992c0)}}, Count:1, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events "centos-4.k8s.com.150bc9f44190cb0a" is forbidden: not yet ready to handle request' (will not retry!)
Jan 21 09:45:24 centos-4.k8s.com kubelet[4857]: I0121 09:45:24.340531    4857 kubelet_node_status.go:273] Setting node annotation to enable volume controller attach/detach
Jan 21 09:45:31 centos-4.k8s.com kubelet[4857]: E0121 09:45:30.632510    4857 eviction_manager.go:238] eviction manager: unexpected err: failed to get node info: node "centos-4.k8s.com" not found
Jan 21 09:45:38 centos-4.k8s.com kubelet[4857]: W0121 09:45:38.114159    4857 cni.go:171] Unable to update cni config: No networks found in /etc/cni/net.d
Jan 21 09:45:38 centos-4.k8s.com kubelet[4857]: I0121 09:45:38.209237    4857 kubelet_node_status.go:273] Setting node annotation to enable volume controller attach/detach
Jan 21 09:45:38 centos-4.k8s.com kubelet[4857]: E0121 09:45:38.243752    4857 kubelet.go:2105] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
Jan 21 09:45:38 centos-4.k8s.com kubelet[4857]: I0121 09:45:38.486750    4857 kubelet_node_status.go:82] Attempting to register node centos-4.k8s.com
Jan 21 09:45:45 centos-4.k8s.com kubelet[4857]: E0121 09:45:45.544591    4857 eviction_manager.go:238] eviction manager: unexpected err: failed to get node info: node "centos-4.k8s.com" not found

**What you expected to happen**: kubeadm init should complete

**How to reproduce it (as minimally and precisely as possible)**: ESX VM with CentOS 7.3 and kubeadm 1.9.2

**Anything else we need to know?**: Kubelet is working but kubeadm is not completing.

**Environment**:  CentOS 7.3 on ESX VM, Docker 1.12.6
- Kubernetes version (use `kubectl version`):kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T10:09:24Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
- Cloud provider or hardware configuration: ESX VM
- OS (e.g. from /etc/os-release):CentOS Linux release 7.3.1611 (Core)
- Kernel (e.g. `uname -a`):Linux centos-4.k8s.com 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Others: System doesn't have internet access, proxy access only.

**Comment by muralikanagala** [comment]
What is the version of gcr.io/google_containers/etcd-amd64 docker image you are running?
I have faced a similar issue and found that the latest version of that docker image 3.1.11 is listening on wrong IP address. 
I have fixed it by pulling an older version of it from dockerhub (https://hub.docker.com/r/fishead/gcr.io.google_containers.etcd-amd64/~/dockerfile/) and labeling it as latest version.

**Comment by pbedadham** [comment]
I am using 3.1.11 and tried with 3.1.10 but no luck.
----
docker images
REPOSITORY                                               TAG                 IMAGE ID            CREATED             SIZE
gcr.io/google_containers/kube-apiserver-amd64            v1.9.2              7109112be2c7        3 days ago          210.4 MB
gcr.io/google_containers/kube-controller-manager-amd64   v1.9.2              769d889083b6        3 days ago          137.8 MB
gcr.io/google_containers/kube-scheduler-amd64            v1.9.2              2bf081517538        3 days ago          62.71 MB
gcr.io/google_containers/etcd-amd64                      3.1.10              1406502a6459        4 months ago        192.7 MB
gcr.io/google_containers/etcd-amd64                      3.1.11              1406502a6459        4 months ago        192.7 MB
gcr.io/google_containers/pause-amd64                     3.0                 99e59f495ffa        20 months ago       746.9 kB
-------

**Comment by pbedadham** [comment]
Need help from Iptables experts- Anything wrong and the below iptables. Just wanted to be clear that my issue is not related to Iptables.
iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
KUBE-FIREWALL  all  --  anywhere             anywhere

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
KUBE-FIREWALL  all  --  anywhere             anywhere

Chain DOCKER (0 references)
target     prot opt source               destination

Chain DOCKER-ISOLATION (0 references)
target     prot opt source               destination

Chain KUBE-FIREWALL (2 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere             /* kubernetes firewall for dropping marked packets */ mark match 0x8000/0x8000

**Comment by muralikanagala** [comment]
I think this issue is related to this.
https://github.com/kubernetes/heapster/issues/1804

etcd container should be listening on localhost (127.0.0.1) but your isp may resolve it to some weird ip.
I will try using the latest etcd image. The old one probably is hard coded to use 127.0.0.1.

**Comment by pbedadham** [comment]
Yeah- Some thing wrong with my network resolution or image. Spun a new VM with new image and worked as expected. 

Thank you for your support.

**Comment by jasonrooks** [comment]
I've been running into this a lot and just occurred again.  I'm trying to install on CentOS 7. 
On a fresh CentOS 7 install preformed the following steps
`kubeadm init`

It ran as expected and I was able to run
`kubectl get pods`

I decided I wanted to use flannel and saw that I needed to use
`kubeadm init --pod-network-cidr=10.244.0.0/16`
instead so I ran 
`kubeadm reset`
then
`kubeadm init --pod-network-cidr=10.244.0.0/16`
It again ran successfully but then I could not use kubectl.  Every attempt returned
kubectl get pods
Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")`

I again tried 
`kubeadm reset`
then
`kubeadm init --pod-network-cidr=10.244.0.0/16`
and now kubeadm init just hangs.
The CA Authority has been corrupted or something.  
This is my docker log for the kube-apiserver containter
`I0206 20:28:56.556075       1 server.go:121] Version: v1.9.2
I0206 20:28:56.973501       1 feature_gate.go:184] feature gates: map[Initializers:true]
I0206 20:28:56.973535       1 initialization.go:90] enabled Initializers feature as part of admission plugin setup
I0206 20:28:56.981050       1 master.go:225] Using reconciler: master-count
W0206 20:28:57.028530       1 genericapiserver.go:342] Skipping API batch/v2alpha1 because it has no resources.
W0206 20:28:57.041591       1 genericapiserver.go:342] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0206 20:28:57.044810       1 genericapiserver.go:342] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0206 20:28:57.066974       1 genericapiserver.go:342] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/02/06 20:28:57 log.go:33: [restful/swagger] listing is available at https://10.1.4.141:6443/swaggerapi
[restful] 2018/02/06 20:28:57 log.go:33: [restful/swagger] https://10.1.4.141:6443/swaggerui/ is mapped to folder /swagger-ui/
[restful] 2018/02/06 20:28:57 log.go:33: [restful/swagger] listing is available at https://10.1.4.141:6443/swaggerapi
[restful] 2018/02/06 20:28:57 log.go:33: [restful/swagger] https://10.1.4.141:6443/swaggerui/ is mapped to folder /swagger-ui/
I0206 20:29:01.328210       1 serve.go:89] Serving securely on [::]:6443
I0206 20:29:01.328278       1 apiservice_controller.go:112] Starting APIServiceRegistrationController
I0206 20:29:01.328299       1 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I0206 20:29:01.328384       1 crd_finalizer.go:242] Starting CRDFinalizer
I0206 20:29:01.329640       1 available_controller.go:262] Starting AvailableConditionController
I0206 20:29:01.329653       1 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I0206 20:29:01.329676       1 controller.go:84] Starting OpenAPI AggregationController
I0206 20:29:01.330114       1 crdregistration_controller.go:110] Starting crd-autoregister controller
I0206 20:29:01.330128       1 controller_utils.go:1019] Waiting for caches to sync for crd-autoregister controller
I0206 20:29:01.330154       1 customresource_discovery_controller.go:152] Starting DiscoveryController
I0206 20:29:01.330180       1 naming_controller.go:274] Starting NamingConditionController
I0206 20:29:35.442035       1 trace.go:76] Trace[778962578]: "Create /api/v1/nodes" (started: 2018-02-06 20:29:01.440687092 +0000 UTC m=+5.034877409) (total time: 34.001307054s):
Trace[778962578]: [4.000941983s] [4.000753339s] About to store object in database
Trace[778962578]: [34.001307054s] [30.000365071s] END
I0206 20:29:36.334779       1 trace.go:76] Trace[162524481]: "Create /api/v1/namespaces" (started: 2018-02-06 20:29:02.333586936 +0000 UTC m=+5.927777264) (total time: 34.001158889s):
Trace[162524481]: [4.000760239s] [4.000680394s] About to store object in database
Trace[162524481]: [34.001158889s] [30.00039865s] END
E0206 20:29:36.335342       1 client_ca_hook.go:78] Timeout: request did not complete within allowed duration
E0206 20:30:01.442817       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *rbac.RoleBinding: the server was unable to return a response in the time allotted, but may still be processing the request (get rolebindings.rbac.authorization.k8s.io)
E0206 20:30:01.442926       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.ResourceQuota: the server was unable to return a response in the time allotted, but may still be processing the request (get resourcequotas)
E0206 20:30:01.443351       1 reflector.go:205] k8s.io/kubernetes/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1.Endpoints: the server was unable to return a response in the time allotted, but may still be processing the request (get endpoints)
E0206 20:30:01.443957       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *rbac.Role: the server was unable to return a response in the time allotted, but may still be processing the request (get roles.rbac.authorization.k8s.io)
E0206 20:30:01.444014       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Secret: the server was unable to return a response in the time allotted, but may still be processing the request (get secrets)
E0206 20:30:01.446161       1 reflector.go:205] k8s.io/kubernetes/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1.Namespace: the server was unable to return a response in the time allotted, but may still be processing the request (get namespaces)
E0206 20:30:01.447206       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *storage.StorageClass: the server was unable to return a response in the time allotted, but may still be processing the request (get storageclasses.storage.k8s.io)
E0206 20:30:01.455492       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *rbac.ClusterRoleBinding: the server was unable to return a response in the time allotted, but may still be processing the request (get clusterrolebindings.rbac.authorization.k8s.io)
E0206 20:30:01.456110       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.PersistentVolume: the server was unable to return a response in the time allotted, but may still be processing the request (get persistentvolumes)
E0206 20:30:01.456521       1 reflector.go:205] k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/client/informers/internalversion/factory.go:73: Failed to list *apiextensions.CustomResourceDefinition: the server was unable to return a response in the time allotted, but may still be processing the request (get customresourcedefinitions.apiextensions.k8s.io)
E0206 20:30:01.456909       1 reflector.go:205] k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/client/informers/internalversion/factory.go:73: Failed to list *apiregistration.APIService: the server was unable to return a response in the time allotted, but may still be processing the request (get apiservices.apiregistration.k8s.io)
E0206 20:30:01.457447       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Pod: the server was unable to return a response in the time allotted, but may still be processing the request (get pods)
E0206 20:30:01.457503       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *rbac.ClusterRole: the server was unable to return a response in the time allotted, but may still be processing the request (get clusterroles.rbac.authorization.k8s.io)
E0206 20:30:01.463815       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.ServiceAccount: the server was unable to return a response in the time allotted, but may still be processing the request (get serviceaccounts)
E0206 20:30:01.464722       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.LimitRange: the server was unable to return a response in the time allotted, but may still be processing the request (get limitranges)
E0206 20:30:01.465071       1 reflector.go:205] k8s.io/kubernetes/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1.Service: the server was unable to return a response in the time allotted, but may still be processing the request (get services)
E0206 20:30:02.329856       1 storage_rbac.go:175] unable to initialize clusterroles: the server was unable to return a response in the time allotted, but may still be processing the request (get clusterroles.rbac.authorization.k8s.io)
I0206 20:30:06.336375       1 trace.go:76] Trace[1836040073]: "Create /api/v1/namespaces" (started: 2018-02-06 20:29:36.335902689 +0000 UTC m=+39.930093003) (total time: 30.0004453s):
Trace[1836040073]: [30.0004453s] [30.000354111s] END
E0206 20:30:06.336872       1 client_ca_hook.go:78] Timeout: request did not complete within allowed duration
F0206 20:30:06.336895       1 hooks.go:188] PostStartHook "ca-registration" failed: unable to initialize client CA configmap: timed out waiting for the condition`
